### PR TITLE
feat: atlas-basic module local tests

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description
This PR contains the local tests for the atlas-basic module, as well as a README that explains how to execute locally such tests. 
Specifically, there are 3 tests:
- atlas_basic_complete: a test where all the input variables are given
- atlas_basic_empty: a test where none of the input variables are given
- atlas_basic_tenant: a test where a TENANT cluster is created

## Important
- The tests (in total) last 30+ minutes to execute (this is because of the create cluster resource)
- The random_name_generator "submodule" has been added to generate a random project name at each test and execution

### CODE REVIEW

**Comment by oarbusi** [comment]
Could you put an image that shows that the tests have been run successfully? (for future reference)
Might be out of scope of this PR, but it would be great that the tests run in the CI, maybe in a follow up PR? is this planned @AgustinBettati @rubenVB01 ?

**Comment by rubenVB01** [comment]
> Could you put an image that shows that the tests have been run successfully? (for future reference) Might be out of scope of this PR, but it would be great that the tests run in the CI, maybe in a follow up PR? is this planned @AgustinBettati @rubenVB01 ?

Yes, I upload here an image. The CI aspect it will be done in a future ticket / PR, yes. 

Screenshot 2024-07-18 at 16 23 17

**Review by EspenAlbert — APPROVED** [review]
Nice!

File: tests/atlas_basic_complete.tftest.hcl [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      {
+        username = "user1"
+        password = "1234"
+        roles = [
+          {
+            role     = "atlasAdmin"
+            database = "admin"
+          }
+        ]
+        scopes = [
```
Comment (AgustinBettati) [inline tests/atlas_basic_complete.tftest.hcl:35]: could we add assertion for scopes and roles?

File: tests/atlas_basic_complete.tftest.hcl [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      }
+    ]
+
+    electable_specs = {
+      instance_size = "M10"
+      node_count    = 5
+    }
+    analytics_specs = {
+      instance_size = "M10"
+      node_count    = 3
```
Comment (AgustinBettati) [inline tests/atlas_basic_complete.tftest.hcl:60]: would also be good to assert the node count of analytics_specs

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+## Instructions on how to run the terraform modules tests
+
+1. The Terraform version should be the latest (i.e. v1.9.0): this is advised because on earlier versions, tests were at beta stage.
+2. Through terminal, set up the following environment variables: 
+    -  `export MONGODB_ATLAS_PUBLIC_KEY="<YOUR_PUBLIC_KEY>"`
+    -  `export MONGODB_ATLAS_PRIVATE_KEY="<YOUR_PRIVATE_KEY>"`
+    -  `export MONGODB_ATLAS_BASE_URL="https://cloud-dev.mongodb.com/"` 
+    -  `export TF_VAR_org_id=<YOUR_ORG_ID>` 
+3. Run from terminal (from repository root) the command `terraform init`
+4. Run from terminal (from repository root) the command `terraform test`
```
Comment (AgustinBettati) [inline tests/README.md:10]: [q] As tests are defined in separate files, do they run in parallel?

**Review by AgustinBettati — APPROVED** [review]
LGTM! Leaving some improvement suggestions and doubts

File: tests/README.md (hunk 3 above)
Comment (rubenVB01) [reply tests/README.md:10]: No, they actually run sequentially like this:
1. Terraform runs a certain test (its run blocks) and creates the corresponding plan, resources, and state.
2. Once everything is created and the asserts are passed (i.e. the test pass correctly), Terraform destroys everything that was created within the test
3. Then, Terraform runs the next test

Comment (AgustinBettati) [reply tests/README.md:10]: thanks for the details, just saw they have an open issue as well related to this: https://github.com/hashicorp/terraform/issues/34180

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Pull request merged**